src/TLuaInterpreter.cpp

```diff
@@ -10042,102 +10067,119 @@ int TLuaInterpreter::setGridMode(lua_State* L)
-        lua_pushstring(L, "setFgColor: wrong argument type");
-        lua_error(L);
-        return 1;
+    if (!lua_isnumber(L, ++s)) {
+        lua_pushfstring(L, "setFgColor: bad argument #%d type (green component value as number expected, got %s!)", s, luaL_typename(L, s));
+        return lua_error(L);
     } else {
         luaGreen = lua_tointeger(L, s);
-        s++;
+        if (luaGreen< 0 || luaGreen >  255) {
+            lua_pushnil(L);
+            lua_pushfstring(L, "the color's green component value %d is outside of the valid range (0 to 255)", luaGreen);
+            return lua_error(L);
+        }
     }
 
-    if (!lua_isnumber(L, s)) {
-        lua_pushstring(L, "setFgColor: wrong argument type");
-        lua_error(L);
-        return 1;
+    if (!lua_isnumber(L, ++s)) {
+        lua_pushfstring(L, "setFgColor: bad argument #%d type (blue component value as number expected, got %s!)", s, luaL_typename(L, s));
+
+        return lua_error(L);
     } else {
         luaBlue = lua_tointeger(L, s);
+        if (luaBlue < 0 || luaBlue >  255) {
+            lua_pushnil(L);
+            lua_pushfstring(L, "the color's blue component value %d is outside of the valid range (0 to 255)", luaBlue);
+            return lua_error(L);
+        }
     }
 
-    QString _name(a1.c_str());
     Host& host = getHostFromLua(L);
-    if (n < 4) {
-        host.mpConsole->setFgColor(luaRed, luaGreen, luaBlue);
-    } else {
-        mudlet::self()->setFgColor(&host, _name, luaRed, luaGreen, luaBlue);
-    }
+
+    mudlet::self()->setFgColor(&host, windowName, luaRed, luaGreen, luaBlue);
     return 0;
 }
 
 // Documentation: https://wiki.mudlet.org/w/Manual:Lua_Functions#setBgColor
 int TLuaInterpreter::setBgColor(lua_State* L)
 {
-    int s = 1;
+    int s = 0;
     int n = lua_gettop(L);
-    string a1;
+    QString windowName = QLatin1String("main");
     int luaRed;
     int luaGreen;
     int luaBlue;
     if (n > 3) {
-        if (lua_isstring(L, s)) {
-            a1 = lua_tostring(L, s);
-            s++;
+        if (!lua_isstring(L, ++s)) {
+            lua_pushfstring(L, "setBgColor: bad argument #%d type (window name as string expected, got %s!)", s, luaL_typename(L, s));
+            return lua_error(L);
+        } else {
+            windowName = QString::fromUtf8(lua_tostring(L, s));
         }
     }
-    if (!lua_isnumber(L, s)) {
-        lua_pushstring(L, "setBgColor: wrong argument type");
-        lua_error(L);
-        return 1;
+    if (!lua_isnumber(L, ++s)) {
+        lua_pushfstring(L, "setBgColor: bad argument #%d type (red component value as number expected, got %s!)", s, luaL_typename(L, s));
+        return lua_error(L);
     } else {
         luaRed = lua_tointeger(L, s);
-        s++;
+        if (luaRed < 0 || luaRed >  255) {
+            lua_pushnil(L);
+            lua_pushfstring(L, "the color's red component value %d is outside of the valid range (0 to 255)", luaRed);
+            return lua_error(L);
```

Comment:
:stop_sign:  Ah, this needs to be `return 2;` it does not want to `return lua error(L);` as it is not a problem with the *type* of the argument (which means the script that called it is broken) but instead we want to signal that the value supplied is out of range which is a less severe error and one which a script can recover from - if it is set to check for a `nil` return value and display the associated error message {using lua_error aborts the **calling function** but returning a `nil` allows it to keep running and to try and handle the error}!!!

Reply:
applied retroactively across the board